src/en/hentairead/src/eu/kanade/tachiyomi/extension/en/hentairead/Hentairead.kt

```diff
@@ -50,31 +91,108 @@ class Hentairead : Madara("HentaiRead", "https://hentairead.com", "en", dateForm
-
-    override val pageListParseSelector = "li.chapter-image-item > a > div.image-wrapper"
+    override fun popularMangaRequest(page: Int): Request = GET("$baseUrl/$mangaSubString/${searchPage(page)}?sortby=views", headers)
+    override fun latestUpdatesRequest(page: Int): Request = GET("$baseUrl/$mangaSubString/${searchPage(page)}?sortby=new", headers)
+    override fun popularMangaSelector() = ".manga-item"
+    override val popularMangaUrlSelector = ".manga-item__bottom a"
+
+    private fun getTagId(tag: String, type: String): Int? {
+        val ajax = "https://hentairead.com/wp-admin/admin-ajax.php?action=search_manga_terms&search=$tag&taxonomy=$type".replace("artist", "manga_artist")
+        val res = client.newCall(GET(ajax, headers)).execute()
+        val items = res.parseAs<Results>()
+        val item = items.results.filter { it.text.lowercase() == tag.lowercase() }
+        if (item.isNotEmpty()) {
+            return item[0].id
+        }
+        return null
+    }
+    override fun searchMangaParse(response: Response): MangasPage = popularMangaParse(response)
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = baseUrl.toHttpUrl().newBuilder().apply {
+            addPathSegments("page/$page")
+            addQueryParameter("s", query)
+            addQueryParameter("title-type", "contains")
+            filters.forEach {
+                when (it) {
+                    is TypeFilter -> {
+                        val (activeFilter, inactiveFilters) = it.state.partition { stIt -> stIt.state }
+                        activeFilter.map { fil -> addQueryParameter("categories[]", fil.value) }
+                    }
+
+                    is PageFilter -> {
+                        if (it.state.isNotBlank()) {
+                            val (min, max) = parsePageRange(it.state)
+                            addQueryParameter("pages", "$min-$max")
+                        }
+                    }
+
+                    is UploadedFilter -> {
+                        if (it.state.isNotBlank()) {
+                            val type = when (it.state.firstOrNull()) {
+                                '>' -> "after"
+                                '<' -> "before"
+                                else -> "in"
+                            }
+                            addQueryParameter("release-type", type)
+                            addQueryParameter("release", it.state.filter(Char::isDigit))
+                        }
+                    }
+
+                    is TextFilter -> {
+                        if (it.state.isNotEmpty()) {
+                            it.state.split(",").filter(String::isNotBlank).map { tag ->
+                                val trimmed = tag.trim()
+                                val id = getTagId(trimmed.removePrefix("-"), it.type)?.toString()
```

Comment:
would this make n calls for n tags?

Reply:
> would this make n calls for n tags?

Yes, I have yet to find a way to get all of the tags' ids at once.